webapp/components/compat-2021.js

```diff
@@ -44,14 +113,73 @@ class Compat2021 extends PolymerElement {
   static get properties() {
     return {
       stable: Boolean,
+      feature: String,
     };
   }
 
+  static get observers() {
+    return [
+      'updateUrlParams(stable, feature)',
+    ];
+  }
+
   ready() {
     super.ready();
 
     const params = (new URL(document.location)).searchParams;
     this.stable = params.get('stable') !== null;
+    this.feature = params.get('feature');
+
+    // The default behavior of the page (when loaded with no params) is to not
+    // select any graph, so we can directly set `value` from the param here.
+    this.$.featureSelect.value = this.feature;
+
+    this.$.featureSelect.addEventListener('change', () => {
+      this.feature = this.$.featureSelect.value;
+    });
+  }
+
+  updateUrlParams(stable, feature) {
+    // Our observer may be called before the feature is set, so debounce that.
+    if (feature === undefined) {
+      return;
+    }
+
+    const params = [];
+    if (feature) {
+      params.push(`feature=${feature}`);
+    }
+    if (stable) {
+      params.push('stable');
+    }
+
+    // We always append a '?' at the very least, as passing empty-string to
```

Comment:
If we use the full URL, or at least prepend `/compat2021`, can we avoid appending the `?`?

I think it's just the empty string that's special cased.

Reply:
Ah, excellent. `location.pathname` works, thanks!